File: Submissions/MailRuCup_2018_Round1/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int x = sc.nextInt();
    int y = sc.nextInt();
    int z = sc.nextInt();
    int t1 = sc.nextInt();
    int t2 = sc.nextInt();
    int t3 = sc.nextInt();
    sc.close();

    String answer = "YES";

    int d1 = Math.abs(x - y);
    int d2 = Math.abs(x - y) + Math.abs(x - z);

    if (d1 * t1 < d2 * t2 + t3 * 3) {
      answer = "NO";
    }

    System.out.println(answer);
    
  }

}

File: Submissions/MailRuCup_2018_Round1/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int x, y, z, t1, t2, t3;
  cin >> x >> y >> z >> t1 >> t2 >> t3;

  string answer = "YES";

  int d1 = abs(x - y);
  int d2 = abs(x - y) + abs(x - z);

  if (d1 * t1 < d2 * t2 + t3 * 3) {
    answer = "NO";
  }
  
  cout << answer << endl;

}